server/py/services/api/main.py

```diff
@@ -982,10 +985,33 @@ async def _retry_jobs(self):
                 for run_dict in runs:
                     run = mlrun.RunObject.from_dict(run_dict)
                     if run.metadata.uid in self._retry_in_progress_run_uids:
-                        self._logger.debug(
-                            "Run is already being retried, skipping",
-                            run_uid=run.metadata.uid,
-                        )
+                        first_retry_time = self._retry_in_progress_run_uids[
+                            run.metadata.uid
+                        ]
+                        if now - first_retry_time > stale_after:
+                            self._logger.warning(
+                                "Run is stale, aborting retry",
+                                run_uid=run.metadata.uid,
+                                first_retry_time=first_retry_time,
+                                now=now,
+                            )
+                            futures.append(
+                                fastapi.concurrency.run_in_threadpool(
+                                    framework.db.session.run_function_with_new_db_session,
+                                    services.api.crud.Runs().abort_run,
+                                    project=run.metadata.project,
+                                    uid=run.metadata.uid,
+                                    run_updates={
+                                        "status.status_text": f"Run retry timed out after {staleness_threshold} days",
```

Comment:
maybe it is better to include info about it being aborted, something like:
```suggestion
                                        "status.status_text": f"Retry aborted: run was pending retry for more than {staleness_threshold} days",
```